docs/articles/documentation/test-api/identify-the-browser-and-platform.md

```diff
@@ -0,0 +1,296 @@
+
+```js
+{
+    alias: 'saucelabs:Samsung Galaxy S9 Plus WQHD GoogleAPI Emulator@8.1'
+}
+```
+
+### name
+
+The short browser name.
+
+```js
+{
+    name: 'Safari'
+}
+```
+
+```js
+{
+    name: 'Internet Explorer'
+}
+```
+
+If the browser cannot be [detected automatically](../using-testcafe/common-concepts/browsers/browser-support.md#locally-installed-browsers), the `name` property is set to `'Other'`.
+
+### version
+
+The browser version.
+
+```js
+{
+    name: 'Chrome',
+    version: '77.0.3865.120'
+}
+```
+
+```js
+{
+    name: 'Firefox',
+    version: '69.0'
+}
+```
+
+The `version` property is set to `'0.0'` if the browser version cannot be determined.
+
+### platform
+
+Identifies the platform type. This property can have the following values:
+
+* `desktop`
+* `mobile`
+* `tablet`
+* `other` (identifies other platforms, or indicates that the platform cannot be detected)
+
+```js
+{
+    name: 'Firefox',
+    platform: 'mobile'
+}
+```
+
+### headless
+
+Specifies if the browser is in [headless mode](../using-testcafe/common-concepts/browsers/testing-in-headless-mode.md).
+
+```js
+{
+    alias: 'chrome:headless',
+    name: 'Chrome',
+    headless: true
+}
+```
+
+### os
+
+Provides the operating system's `name` and `version`.
+
+```js
+{
+    os: {
+        name: 'Windows',
+        version: '10'
+    }
+}
+```
+
+If the operating system cannot be detected, the `os.name` property is set to `'Other'`. The `os.version` property defaults to `'0.0'` if TestCafe is unable to determine the OS version.
+
+### engine
+
+Provides the browser engine's `name` and `version`.
+
+```js
+{
+    engine: {
+        name: 'WebKit',
+        version: '605.1.15'
+    }
+}
+```
+
+If the browser engine cannot be detected, the `engine.name` property is set to `'Other'`. The `engine.version` property defaults to `'0.0'` if TestCafe is unable to determine the engine version.
+
+### userAgent
+
+The user agent string.
+
+```js
+{
+    userAgent: 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/77.0.3865.120 Safari/537.36'
+}
+```
+
+### prettyUserAgent
+
+The formatted name and version of the browser and operating system. This string is displayed on the TestCafe loading screen and in the status bar.
+
+```js
+{
+    prettyUserAgent: 'Firefox 69.0 / Windows 10'
+}
+```
+
+```js
+{
+    prettyUserAgent: 'Chrome 77.0.3865.120 / macOS 10.15.1'
+}
+```
+
+The `prettyUserAgent` property is set to an empty string if TestCafe cannot parse the user agent.
+
+## Examples
+
+### Create a Browser-Specific Hook
+
+The following example shows how to create a [beforeEach](test-code-structure.md#test-hooks) hook that runs for only particular [browser engines](#engine).
```

Comment:
The following example shows how to create a [beforeEach](test-code-structure.md#test-hooks) hook that runs for particular [browser engines](#engine) only: